ckan/config/declaration/option.py

```diff
@@ -363,12 +364,13 @@ def _validators_from_string(s: str) -> list[Validator]:
     from ckan.logic import get_validator
 
     out = []
-    parts = s.split()
+    parts = re.split(r"\s+", s) if s else s.split()
+
     for p in parts:
         if "(" in p and p[-1] == ")":
             name, args = p.split("(", 1)
             args: Any = args[:-1].split(",")  # trim trailing ')', break up
-            v = get_validator(name)(*args)
+            v = get_validator(name)(args)  # type: ignore
```

Comment:
I really don't like it. Yes, you get the ability to use `OneOf`, but you won't be able to use custom validator factories anymore, like:
```python
def create_min_max_validator(min, max):
    def validator(value):
        ...
    return validator 

# declaration
validators: create_min_max_validator(1, 2)
```


And don't forget, that in the python code `OneOf(x, y)` won't work either. it must be `OneOf([x, y])`. You have to pass there a single collection.

So here we have a breaking change + a new behavior. I think that it has more sense to:
 
* parse literals(i.e. allow something like `validators: OneOf(["Lax", "Strict"])`). It's probably too much for now and requires careful planning
* create a separate validator. maybe 
  ```python
  def samesite_value(value):
      return one_of(["Lax", "Strict", "None"])(value)
  ```

Reply:
I thought having specific validator only for `samesite` field is not making much sense but we will need more generic solution to check options with `OneOf` 

I was unaware about the custom factorie and i will try to improve the solution, although spliting the string is the toughest challenge as we could hit more validators e.g `validator1(a,b) validator2 validator3(a)`

Reply:
parsing literals with `ast.literal_eval` should be fine. We can fall back to treating the parameter as a simple string when that fails with a ValueError for backwards compatibility. I'd like to do the same in scheming if you make that change here